test/sshkit/scp_functional_test.exs

```diff
@@ -9,29 +9,141 @@ defmodule SSHKit.SCPFunctionalTest do
   @remote_workspace "/workspace"
   @local_remote_workspace "test/fixtures/docker_workspace"
 
-  @tag boot: 1
-  test "uploads a file", %{hosts: [host]} do
-    options = [port: host.port, user: host.user, password: host.password]
-    local = "#{@local_workspace}/local_file.txt"
-    remote = "#{@remote_workspace}/#{host.id}.file"
-    shared = "#{@local_remote_workspace}/#{host.id}.file"
-
-    {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
-    assert :ok = SCP.upload(conn, local, remote)
-    assert File.read(local) == File.read(shared)
-    File.rm!(shared)
+  describe "upload" do
+
+    @tag boot: 1
+    test "to host directly", %{hosts: [host]} do
+      options = [port: host.port, user: host.user, password: host.password]
+      local = "#{@local_workspace}/local_file.txt"
+      remote = "#{@remote_workspace}/#{host.id}.file"
+      shared = "#{@local_remote_workspace}/#{host.id}.file"
+
+      {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
+      assert :ok = SCP.upload(conn, local, remote)
+      assert File.read(local) == File.read(shared)
+      File.rm!(shared)
+    end
+
+    @tag boot: 2
+    test "using a context", %{hosts: hosts} do
+      local = "#{@local_workspace}/local_file.txt"
+      remote = "#{@remote_workspace}/destination.file"
+      shared = "#{@local_remote_workspace}/destination.file"
+
+      hosts = Enum.map(hosts,
+        fn(h) ->
+          SSHKit.host(h.ip, Keyword.merge(@defaults, [port: h.port, user: h.user, password: h.password]))
+        end)
+      context = SSHKit.context(hosts)
+
+      assert [:ok, :ok] = SSHKit.upload(context, local, as: remote)
+      assert File.read(local) == File.read(shared)
+      File.rm!(shared)
+    end
```

Comment:
This is really testing `SSHKit.upload/3` so it should probably go into `test/sshkit_functional_test.exs`.